plugins/signals-backend/src/service/router.ts

```diff
@@ -56,16 +58,17 @@ export async function createRouter(
   let apiUrl: string | undefined = undefined;
 
   const webSocketServer = new WebSocketServer({
-    noServer: true,
-    clientTracking: false,
+    noServer: true, // handle upgrade manually
+    clientTracking: false, // handle connections in SignalManager
   });
 
   webSocketServer.on('error', (error: Error) => {
-    logger.error('WebSocket server error', error);
+    logger.error(`WebSocket server error: ${error}`);
   });
 
   webSocketServer.on('close', () => {
     logger.info('WebSocket server closed');
+    manager.onServerClose();
   });
```

Comment:
Hmm, does this ever happen? Would've expected this to be hooked up to the lifecycle service

Reply:
It's better to just hook the `SignalManager` to the lifecycle and let the HTTP server close handle shutting down the websocket server